psh/sysexec/sysexec.c

```diff
@@ -130,35 +138,78 @@ static int psh_sysexec(int argc, char **argv)
 
 			case 'h':
 				psh_sysexecUsage();
-				return EOK;
+				return EXIT_SUCCESS;
+
+			case 'd':
+				background = true;
+				break;
+
+			case 's':
+				keepStdin = true;
+				break;
 
 			default:
 				psh_sysexecUsage();
-				return -EINVAL;
+				return EXIT_FAILURE;
 		}
 	}
 
+	if (!background && keepStdin) {
+		fprintf(stderr, "psh: -s option can be only used with -d option\n");
+		psh_sysexecUsage();
+		return EXIT_FAILURE;
+	}
+
 	progName = argv[optind];
 	if (progName == NULL) {
 		fprintf(stderr, "psh: missing program name for sysexec\n");
 		psh_sysexecUsage();
-		return -EINVAL;
+		return EXIT_FAILURE;
 	}
 
 	if (psh_sysexec_checkcommand(argc, (const char **)argv) != 1) {
 		fprintf(stderr, "Unknown command!\n");
-		return -EINVAL;
+		return EXIT_FAILURE;
+	}
+
+	pid_t pid;
+	if (background) {
+		pid = vfork();
+	}
+	else {
+		pid = spawnSyspage(codeMapName, dataMapName, progName, &argv[optind]);
 	}
 
-	pid = spawnSyspage(codeMapName, dataMapName, progName, &argv[optind]);
-	if (pid > 0) {
+	/* Forked path */
+	if (pid == 0) {
+		/* Create a new SID for the child process */
+		int sid = setsid();
```

Comment:
because of [libphoenix/include/unistd.h#L229](https://github.com/phoenix-rtos/libphoenix/blob/master/include/unistd.h#L229)
```suggestion
		pid_t sid = setsid();
```